xcell/tests/test_tosacc.py

```diff
@@ -0,0 +1,293 @@
+        elif tr.quantity in ['cmb_convergence']:
+            assert isinstance(tr, sacc.tracers.MapTracer)
+            assert tr.ell == m.get_ell()
+            # Only here because tr.spin is not an attribute of NZTracers
+            assert tr.spin == m.get_spin()
+            assert tr.beam_extra['nl'] == m.get_nl_coupled()[0]
+            assert tr.beam == np.ones_like(tr.ell)
+        else:
+            raise ValueError('Tracer not implemented')
+
+
+@pytest.mark.parametrize('use', ['cls', 'nl', 'fiducial'])
+def test_ell_cl_autocov(use):
+    s = get_sfile(use)
+    data = get_data()
+
+    for dtype in s.s.get_data_types():
+        for trs in s.s.get_tracer_combinations(dtype):
+            if dtype in ['cl_00', 'cl_0e', 'cl_ee']:
+                ix = 0
+            elif dtype in ['cl_0b', 'cl_eb']:
+                ix = 1
+            elif dtype in ['cl_be']:
+                ix = 2
+            elif dtype in ['cl_bb']:
+                ix = 3
+            else:
+                raise ValueError(f'data_type {dtype} must be weird!')
+
+            if use == 'fiducial':
+                cl_class = ClFid(data.data, *trs)
+            else:
+                cl_class = Cl(data.data, *trs)
+
+            if use == 'nl':
+                ell, cl = s.s.get_ell_cl(dtype, trs[0], trs[1])
+                ell2, cl2 = cl_class.get_ell_nl()
+            else:
+                ell, cl, cov = s.s.get_ell_cl(dtype, trs[0], trs[1],
+                                              return_cov=True)
+                ell2, cl2 = cl_class.get_ell_cl()
+
+                cov_class = Cov(data.data, *trs, *trs)
+                nbpw = ell.size
+                ncls = cl2.shape[0]
+                cov2 = cov_class.get_covariance()
+                cov2 = cov2.reshape((nbpw, ncls, nbpw, ncls))[:, ix, :, ix]
+                assert np.max(np.abs((cov - cov2) / np.mean(cov))) < 1e-5
+
+            if use == 'fiducial':
+                # Matrices to bin the fiducial Cell
+                ws_bpw = np.zeros((ell.size, ell2.size))
+                ws_bpw[np.arange(ell.size), ell.astype(int)] = 1
+                assert np.all(cl == ws_bpw.dot(cl2[ix]))
+            else:
+                assert np.all(cl == cl2[ix])
+                assert np.all(ell == ell2)
+
+
+def test_get_dof_tracers():
+    s = get_sfile()
+    for tr1, tr2 in s.s.get_tracer_combinations():
+        s1 = np.max((s.data.get_mapper(tr1).get_spin(), 1))
+        s2 = np.max((s.data.get_mapper(tr2).get_spin(), 1))
+
+        dof = s1 * s2
+        assert dof == s.get_dof_tracers((tr1, tr2))
+
+
+def test_get_datatypes_from_dof():
+    s = get_sfile()
+    assert s.get_datatypes_from_dof(1) == ['cl_00']
+    assert s.get_datatypes_from_dof(2) == ['cl_0e', 'cl_0b']
+    assert s.get_datatypes_from_dof(4) == ['cl_ee', 'cl_eb', 'cl_be', 'cl_bb']
+    with pytest.raises(ValueError):
+        s.get_datatypes_from_dof(3)
+
+
+@pytest.mark.parametrize('m_marg', [True, False])
```

Comment:
is there anything specific about `m_marg` that `to_sacc.py` knows about? I thought it was agnostic to where the covariance came from. If so, I'd just test for the simple case to save time.

Reply:
This was for the case when you already have your Gaussian covariance and what to compute only de marginalized one, so that you can add it later by hand. The only difference in the code is an `if` statement. I'll remove it.

Reply:
ok, I see. Up to you!

Reply:
I've left the function untouched, but changed the decorator so that it only runs with `False`.